### DISCUSSION TITLE
Absorbtion at 55,2V for endless time?

### DISCUSSION THREAD

**Discussion by geromd** [Louisvdw/dbus-serialbattery#170] [Q&A] [answered]
Hi,

I have a question about the voltage adjustmenst in VEConfig and the different charging modes, that Victron is following due to the algorythm.

Before I installed the driver, Victron was charging my 16S 280Ah EVE LiFePO4 cells up to the entered overall voltage (regardless single cell volatage an SOC), and then switched to "Absorption" (holding the 55,2 V and lowering current) for 1 hour and then it was switching to Float, were I entered a significant lower voltage to gave the cells a recovering to their non-load voltage, to not hold them stressed on 55,2V all the time.

Now, with the installed driver, the charging reach 55,2V and go into Absobtion Mode, were the voltage of 55,2V is kept for all the time. The Victron is not going into Float any more and don't reach full battery status any more . 
 
![Absorbtion](https://user-images.githubusercontent.com/109993803/182401148-2c09c834-3b9a-4038-b6fd-d44ca7b7bc5d.jpg)

My settings in VEConfig are as follows:
![Config1](https://user-images.githubusercontent.com/109993803/182404573-2403982e-950e-4078-aad2-c00d55468c3a.jpg)

![Config2](https://user-images.githubusercontent.com/109993803/182404588-d27249d3-928d-4066-ba3a-a0f89d8cddda.jpg)

![Config 3](https://user-images.githubusercontent.com/109993803/182404605-cb72c955-4360-4e5d-afba-6114ab05bf73.jpg)

Is this OK or should I enter any other parameter? 
Does the driver overwrite the 1 hour Absorbtion setting?
Is it OK to keep the LiFePO4 on 55,2V all the time?

**Comment by Louisvdw** [comment, marked as answer]
Hi @geromd 
There are different layers of config in the Victron system.
Each device has their own setting, and then when you add a GX device and connect them the GX will control everything else. The battery will then ask the GX for what it needs and this will override everything else (this is what DVCC is for).

The charge curve is something that is used for lead acid batteries. They don't have a BMS, so the chargers had evolved to include some logic to add curves and stuff. As the lithium batteries all have a BMS that manage each cell this is not used. There are no more float and absorption voltages. Lithiums don't work like that. 
The BMS will now handle all of that.

You question about the 100% SOC, the difference would be what the BMS states the SOC to be. This can be influence by the formulates and logic that your BMS use. Before the driver the system would not know what the BMS thought and would have estimated it. If it never reach 100% it would only be an issue if it was not very close to 100%. You can check the settings for your battery and if the capacity is correctly set. But it could also be that there are some cells not up to voltage that is keeping the SOC value lower. Dig a bit into the min/max cell values and check if you see anything strange. The graphs in the advance section in VRM give a lot of great insight into your battery.

**Reply by ganey** [reply]
When swapping from lead acid to lifepo4, do we still need to use VEconfig to tell the GX we're now using a different battery type even though DVCC is enabled?

**Reply by Louisvdw** [reply]
Yes I would suggest you do make the changes using VEconfig as well. When something does happen and your GX is offline the inverter will revert to those old settings. As long as they are lower voltages it will not break anything and should be fine, but I would suggest you change those to be sure.
In recent firmwares on the MPPT they will stop production if they loose connection to the BMS via the GX. It could be that something like that has been added to the inverter firmwares are well. I have not seen it on the Multiplus II, while I have seen the MPPT doing that.

**Comment by Dr-Gigavolt** [comment]
Hello, I have a similar issue. My system is still under construction, the Multis are not active yet. Only Smart Solar and one of two batteries with JK BMS, the ESS is not configured yet. If the Smart Solar runs stand-alone, the sequence bulk-absorption-float works. With the battery driver (beta 0.14) it never goes to float.

FLOAT_CELL_VOLTAGE = 2.5
MAX_VOLTAGE_TIME_SEC = 15*60
SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90

According this settings in utils.py it should go to float after 15min? Or have I misunderstood something?

**Reply by Louisvdw** [reply]
Make sure in utils.py you have
>CVCM_ENABLE = True

Just a note of float and absorbtion. Those are things we used for Load Acid batteries. Lithium batteries don't really use that. You now have a BMS that manage your cells much better.

**Reply by Dr-Gigavolt** [reply]
CCVM is enabled. Something else to check?

About your note ... I experienced that the absorption has some advantage even for lithium (I have LTO). If all cells are at 2.4 - 2.5V, they are almost full, but with some differencies, even the voltages are equal. If set the absorption to n×2.6V, the ballancer can continue to work due to voltage increase of cells that are at 100% SoC.

**Reply by Dr-Gigavolt** [reply]
BTW, what do you mean "BMS that manage your cells much better"?

The BMS gives just a protection against under/over-voltage of each cell and if it happens, it disconnects the battery. And the balancer does its job, but slowly. 2A cell after cell is not much. The management is done by the charger, thanks to the data from the BMS and your driver.

**Reply by Louisvdw** [reply]
With Lead Acid battery had not mangement. You had to use a charger that would guess what is going on inside the battery cells. For this the chargers were changed to use have different charge stages that would (in most cases) prevent the cells being overcharged. These are the float and absorption stages for lead acids. When one cell dies the hole battery would be killed and the charger would not know about it.

With Lithium batteries it is do dangerous to do it that way as they can explode and cause a fire. So the management was instead moved to the BMS that can keep each cell in the battery safe. The BMS wil in fact now protect the battery when one cell gets close to overvoltage. This also evolved into a BMS that can communicate with the charger and tells it what to do which is where this driver comes in. 
The protection for each individual cell that the BMS has is what makes it so much better than lead acid batteries and why lead acids had to use floats.

If the BMS does not allow the charge then the charger can do nothing. According to me that means the BMS is in control.

By using CVCM this only lowers the charge voltage that the battery requests. It will not change any of the lights on the inverter.

**Reply by Dr-Gigavolt** [reply]
That the Li cell must not be overcharged and the BMS cares about it is clear. But the disconnecting by MOSFETs is IMHO not a normal operating state, it is kind of emergency brake. I am not sure if Multis accept such state, or switch off with error message and put the house into darkness.

So your driver does not activate the float modus (indicated by LED on charger)and does CVCM instead? Then what is the function of the float voltage parameter in the utils.py?

**Reply by Louisvdw** [reply]
You do want to avoid the BMS disconnecting the battery. You will have no power if that happens. And the driver will translate the alarm and requests that the BMS give for your charging system. 
The CVCM will simulate something similar to float state if you want that, but there is no way we can set those lights. Those are managed by an "older" part of the inverter and not accessable.

**Reply by Dr-Gigavolt** [reply]
Thanks 😀
